### PR TITLE
*: parser support crc32 conv bit_xor

### PR DESCRIPTION
**Pull request by nolouch**
now user use `syncer` to sync data needs parser support.I will implement crc32 and conv later. .but bit_xor is a aggregation function. may we just support parse it firstly?
PTAL @shenli @coocood @tiancaiamao

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -291,6 +291,9 @@ import (
 	charFunc	"CHAR_FUNC"
 	charLength	"CHAR_LENGTH"
 	characterLength	"CHARACTER_LENGTH"
+	conv        "CONV"
```
Comment (hanfei1991) [inline parser/parser.y]: Keep the format uniform.

**Comment by shenli** [comment]
@nolouch It is better implement crc32 in this PR and support bit_xor in another PR.

File: parser/parser.y [hunk 2]
```diff
@@ -3021,6 +3024,21 @@ FunctionCallNonKeyword:
 			Args: []ast.ExprNode{$3.(ast.ExprNode)},
 		}
 	}
+|	"CONV" '(' Expression ',' Expression ',' Expression ')'
+	{
+		$$ = &ast.FuncCallExpr{
+            FnName: model.NewCIStr($1),
```
Comment (iamxy) [inline parser/parser.y]: please format it.

File: parser/parser.y [hunk 3]
```diff
@@ -3021,6 +3024,21 @@ FunctionCallNonKeyword:
...
+            Args: []ast.ExprNode{$3.(ast.ExprNode), $5.(ast.ExprNode), $7.(ast.ExprNode)},
+		}
+	}
+|	"CRC32" '(' Expression ')'
+	{
+        $$ = &ast.FuncCallExpr{
```
Comment (iamxy) [inline parser/parser.y]: ditto

File: parser/parser.y [hunk 4]
```diff
@@ -3099,6 +3117,10 @@ FunctionCallAgg:
 	{
 		$$ = &ast.AggregateFuncExpr{F: $1, Args: []ast.ExprNode{$4.(ast.ExprNode)}, Distinct: $3.(bool)}
 	}
+|	"BIT_XOR" '(' DistinctOpt Expression ')'
```
Comment (hanfei1991) [inline parser/parser.y]: What's the behaviour when a user use bit_xor? report an error?

Comment (nolouch) [inline parser/parser.y]: report error is ok. Now just need to make sure parser is work on it.

File: expression/builtin_math.go [hunk 5]
```diff
@@ -185,3 +185,13 @@ func builtinRound(args []types.Datum, ctx context.Context) (d types.Datum, err e
 	d.SetFloat64(types.Round(x, dec))
 	return d, nil
 }
+
+// See http://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_conv
+func builtinConv(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
```
Comment (shenli) [inline expression/builtin_math.go]: Please add TODO here and return unimplement error.
I think it is OK to remove the two functions. You only need to make parser happy.

**Comment by nolouch** [comment]
PTAL @shenli @tiancaiamao

**Comment by shenli** [comment]
@nolouch You need to add type infer info and test in plan/typeinferer.go
See: https://github.com/ngaut/builddatabase/blob/master/tidb/builtin.md

File: expression/builtin_math.go [hunk 6]
```diff
@@ -188,10 +189,22 @@ func builtinRound(args []types.Datum, ctx context.Context) (d types.Datum, err e
...
-	return
+	//TODO  implement
+	return d, errors.New("Function unimplement")
 }
 
 //　See http://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_crc32
 func builtinCRC32(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
-	return
+	switch args[0].Kind() {
+	case types.KindNull:
```
Comment (shenli) [inline expression/builtin_math.go]: if args[0].IsNull() {
 return d. nil
}
// xxxx

**Comment by shenli** [comment]
LGTM

**Pull request merged**